lib/Runtime/Library/JavascriptMap.cpp

```diff
@@ -64,6 +64,15 @@ namespace Js
 
         Var iterable = (args.Info.Count > 1) ? args[1] : library->GetUndefined();
 
+        if (mapObject->map != nullptr)
+        {
+            JavascriptError::ThrowTypeErrorVar(scriptContext, JSERR_ObjectIsAlreadyInitialized, _u("Map"), _u("Map"));
+        }
+
+        // ensure mapObject->map is created before trying to fetch the adder function
+        // see github#2747
+        mapObject->map = RecyclerNew(scriptContext->GetRecycler(), MapDataMap, scriptContext->GetRecycler());
```

Comment:
Thanks

---
In reply to: [128404692](https://github.com/Microsoft/ChakraCore/pull/3397#discussion_r128404692) [](ancestors = 128404692)